apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+      pElement.setAttribute('data-l10n-id', 'no-attachment-text');
+    }
+
+    if (message.type === 'mms' && !isNotDownloaded && !noAttachment) { // MMS
+      return this.mmsContentParser(message).then((mmsContent) => {
+        pElement.appendChild(mmsContent);
+        return messageDOM;
+      });
+    }
+
+    return Promise.resolve(messageDOM);
+  },
+
+  mmsContentParser: function conv_mmsContentParser(message) {
+    return new Promise((resolver) => {
+      SMIL.parse(message, (slideArray) => {
+        resolver(this.createMmsContent(slideArray));
+      });
+    });
+  },
+
+  getMessageStatusMarkup: function(status) {
+    return ['read', 'delivered', 'sending', 'error'].indexOf(status) >= 0 ?
+      this.tmpl.messageStatus.prepare({
+        statusL10nId: 'message-delivery-status-' + status
+      }) : '';
+  },
+
+  appendMessage: function conv_appendMessage(message, hidden) {
+    var timestamp = +message.timestamp;
+
+    // look for an old message and remove it first - prevent anything from ever
+    // double rendering for now
+    var messageDOM = document.getElementById('message-' + message.id);
+
+    if (messageDOM) {
+      this.removeMessageDOM(messageDOM);
+    }
+
+    // build messageDOM adding the links
+    return this.buildMessageDOM(message, hidden).then((messageDOM) => {
+      if (this._stopRenderingNextStep) {
+        return;
+      }
+
+      messageDOM.dataset.timestamp = timestamp;
+
+      // Add to the right position
+      var messageContainer = this.getMessageContainer(timestamp, hidden);
+      this._insertTimestampedNodeToContainer(messageDOM, messageContainer);
+
+      if (this.inEditMode) {
+        this.updateSelectionStatus();
+      }
+
+      if (!hidden) {
+        // Go to Bottom
+        this.scrollViewToBottom();
+      }
+    });
+  },
+
+  /**
+   * Inserts DOM node to the container respecting 'timestamp' data attribute of
+   * the node to insert and sibling nodes in ascending order.
+   * @param {Node} nodeToInsert DOM node to insert.
+   * @param {Node} container Container DOM node to insert to.
+   * @private
+   */
+  _insertTimestampedNodeToContainer: function(nodeToInsert, container) {
+    var currentNode = container.firstElementChild,
+        nodeTimestamp = nodeToInsert.dataset.timestamp;
+
+    while (currentNode && nodeTimestamp > +currentNode.dataset.timestamp) {
+      currentNode = currentNode.nextElementSibling;
+    }
+
+    // With this function, "inserting before 'null'" means "appending"
+    container.insertBefore(nodeToInsert, currentNode || null);
+  },
+
+  showChunkOfMessages: function conv_showChunkOfMessages(number) {
+    var elements = this.container.getElementsByClassName('hidden');
+
+    Array.slice(elements, -number).forEach((element) => {
+      element.classList.remove('hidden');
+      if (element.tagName === 'HEADER') {
+        TimeHeaders.update(element);
+      }
+    });
+  },
+
+  showOptions: function conv_showOptions() {
```

Comment:
I think we can just split this method with minimal duplication related to subject handling.